packages/core/src/lib/createSystem.ts

```diff
@@ -82,13 +82,17 @@ export function createSystem(config: KeystoneConfig) {
 
       setWriteLimit(prismaClient, pLimit(config.db.provider === 'sqlite' ? 1 : Infinity));
       setPrismaNamespace(prismaClient, prismaModule.Prisma);
-      prismaClient.$on('beforeExit', async () => {
-        // Prisma is failing to properly clean up its child processes
-        // https://github.com/keystonejs/keystone/issues/5477
-        // We explicitly send a SIGINT signal to the prisma child process on exit
-        // to ensure that the process is cleaned up appropriately.
-        prismaClient._engine.child?.kill('SIGINT');
-      });
+
+      // beforeExit event is not supported by Prisma Data Proxy
+      if (!prismaClient._dataProxy) {
```

Comment:
I'm accessing the internal variable `prismaClient._dataProxy` here because the only alternative to check for dataProxy usage would be:

```js
prismaClient.getEngine().constructor.name === 'DataProxyEngine'
```